src/main/java/com/lalabrand/ecommerce/auth/JwtService.java

```diff
@@ -0,0 +1,71 @@
+package com.lalabrand.ecommerce.auth;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import javax.crypto.SecretKey;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Component
+public class JwtService {
+    @Value("${secret}")
+    private String key;
+
+    @Value("${access.token.expiration.seconds}")
+    private Long accessTokenExpirationSec;
+
+    public boolean validateToken(String token, UserDetails userDetails) {
+        String username = extractName(token);
+        return (username.equals(userDetails.getUsername()) && !isNotExpired(token));
+    }
+
+    private boolean isNotExpired(String token) {
+        return extractExpiration(token).before(new Date());
+    }
+
+    public String extractName(String token) {
+        return extractClaim(token, Claims::getSubject);
```

Comment:
@denomelchenko Maybe it would be better to somehow convert a token to an intermediate object? This way you don't need to parse it every time separately to get expiration, username, etc.